File: 735-asteroid-collision/asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& arr) {
        int n=arr.size();
        vector<int>st;
        for(int a:arr){
            bool distroyed=false;
            while(!st.empty() and a<0 and st.back()>0){
                if(st.back()< abs(a)){
                    st.pop_back();
                }
                else if(st.back()==abs(a)){
                    st.pop_back();
                    distroyed=true;
                    break;
                }
                else{
                    distroyed=true;
                    break;
                }
            }
            if(!distroyed){
                st.push_back(a);
            }
        }
        return st;
    }
};

File: 735-asteroid-collision/asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: list[int]) -> list[int]:
        ans=[]
        for i in asteroids:
            while ans and ans[-1]>0 and i<0:
                last=ans[-1]
                if last == abs(i):
                    ans.pop()
                    break
                elif last < abs(i):
                    ans.pop()
                else:
                    break
            else:
                ans.append(i)
        return ans